zhaquirks/tuya/ts130f.py

```diff
@@ -21,22 +21,17 @@
     OUTPUT_CLUSTERS,
     PROFILE_ID,
 )
-from zhaquirks.tuya import TuyaZBExternalSwitchTypeCluster
+from zhaquirks.tuya import SwitchBackLight, TuyaZBExternalSwitchTypeCluster
 
 ATTR_CURRENT_POSITION_LIFT_PERCENTAGE = 0x0008
 CMD_GO_TO_LIFT_PERCENTAGE = 0x0005
 
 
-class TuyaWithBacklightOnOffCluster(CustomCluster):
-    """TuyaSmartCurtainOnOffCluster: fire events corresponding to press type."""
+class TuyaWithBacklightOnOffCluster(CustomCluster, OnOff):
+    """Tuya Zigbee On Off cluster with extra attributes."""
 
-    cluster_id = OnOff.cluster_id
-
-    LIGHT_MODE_1 = {0x8001: 0}
-    LIGHT_MODE_2 = {0x8001: 1}
-    LIGHT_MODE_3 = {0x8001: 2}
-
-    attributes = {0x8001: ("backlight_mode", t.enum8)}
+    attributes = OnOff.attributes.copy()
+    attributes.update({0x8001: ("backlight_mode", SwitchBackLight)})
```

Comment:
Maybe misunderstanding, but this alone doesn't create the config entity in ZHA. It just exposes the custom attribute and I think it's done correctly now(?)

In ZHA, most Tuya ZHA config entities match per model names (or manufacturer) and should switch to matching via quirk name/class. Same for adding those custom attributes to the initialization (to avoid throwing a warning for trying to initialize an attribute that wasn't found in the cluster definition, because the quirk wasn't applied, but ZHA still matched the model number).

Some of the Tuya quirks also just expose custom attributes without there being config entities in ZHA at all.

Reply:
The same principle is used for switches and there extra attributes and is implanted in tiya INIT also for windows clovers.

Reply:
This one is abusing onoff and really should be done against the manufacturer cluster. Examples are plentiful in the ZHA switch module. There is no reason for this to use the onoff cluster at all. My assumption is this was done at the time to cause a regular switch entity to be created before we had config entities?

Reply:
Example for the Aqara Opple cluster: https://github.com/home-assistant/core/blob/cc4e741cfa21973cae47a0726e83ec2e51f8ba97/homeassistant/components/zha/switch.py#L530

there are many others in the file.

Reply:
My commentary was really around the fact that the onoff cluster isn’t needed at all to support this functionality and that is the core issue for this particular case.

Reply:
> This one is abusing onoff and really should be done against the manufacturer cluster

Ideally, it should, but this is not a fake attribute. It's a real attribute for changing the LED on the device. Tuya always seems to put it on the `OnOff` cluster.
For plugs, it kind of makes sense. For this device, the rest of the `OnOff` cluster isn't used, so it's _not ideal_.
We can't really move where the attribute is?
Aqara nicely puts all/most custom attributes on a manufacturer specific cluster. Tuya doesn't always do that (of course lol).

> My assumption is this was done at the time to cause a regular switch entity to be created before we had config entities?

No, an entity was never created for this. Before the changes in this PR, the custom cluster class did not subclass `OnOff` at all. Only the cluster ID matched. But since the `ep_attribute` was already different, ZHA never created an entity for this.

Reply:
I have trying collecting tuya special cluster commands and attributes but its not all and all is not doing the same on all device types alla tuya standard.
https://github.com/zigpy/zigpy/discussions/823#discussioncomment-1539469